### ISSUE TITLE
Fix doesn't print PRS errors when templating/parsing errors found

### ISSUE THREAD

**Issue by juhoautio** [sqlfluff/sqlfluff#3372]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried to fix a query with sqlfluff and it failed with "templating/parsing errors found".

### Expected Behaviour

I expected fix output to include the PRS error so that I know what part of the query is problematic.

More generally, I would expect the same errors to be printed regardless of whether I run fix or lint.

### Observed Behaviour

No information was printed about what the templating/parsing error was.

Running lint instead of fix on the same query shows the PRS error as expected.

### How to reproduce

`resources/sql/test.sql`:
```sql
SELECT sum(reduce((map_values(my_map_column)), 0, (s, x) -> s + COALESCE(x, 0), s -> s)) as my_map_values_sum
FROM my_table;
```

Try fix:
```
sqlfluff fix -f resources/sql/test.sql --dialect sparksql
```

```
==== finding fixable violations ====
== [resources/sql/test.sql] FAIL
L:   1 | P:  90 | L010 | Keywords must be consistently upper case.
matching: 2it [00:00,  6.40it/s]  [1 templating/parsing errors found]
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [1 unfixable linting violations found]
```

Note how `[1 templating/parsing errors found]` is printed there ^

Try lint instead:
```
sqlfluff lint resources/sql/test.sql --dialect sparksql
```

```
== [resources/sql/test.sql] FAIL
L:   1 | P:  58 |  PRS | Line 1, Position 58: Found unparsable section: '-> s +
                       | COALESCE(x, 0), s -> s'
L:   1 | P:  90 | L010 | Keywords must be consistently upper case.
matching: 2it [00:00,  7.78it/s]All Finished 📜 🎉!
```

Note how the PRS error is now indicated, as expected ^

### Dialect

This doesn't seem to be dialect specific. The behavior difference between fix & lint is similar (PRS error not printed) for example if I try hive dialect instead of sparksql.

### Version

```
sqlfluff, version 0.13.1
```

### Configuration

No custom conf

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
> More generally, I would expect the same errors to be printed regardless of whether I run fix or lint.

Curious, what's your expectation for rules that can't be autofixed?
Currently they don't print either for `fix`, and `PRS` follows the same lines:

```
% echo "SELECT\n    col1,\n    sum(b)\nFROM table1" | sqlfluff lint --dialect ansi -
== [stdin] FAIL                                                                                                                                                      
L:   3 | P:   5 | L013 | Column expression without alias. Use explicit `AS`
                       | clause.
All Finished 📜 🎉!
% echo "SELECT\n    col1,\n    sum(b)\nFROM table1" | sqlfluff fix --dialect ansi -
Unfixable violations detected.                                                                                                                                       
SELECT
    col1,
    sum(b)
FROM table1
```

I think there is a case to list all errors as you say (perhaps especially for `PRS`, `LEX`, and `TMP` errors since they are serious problems), and do recall this being discussed before, but not sure if people would prefer not to see the noise of unfixable errors.

**Comment by tunetheweb** [comment]
> I would at least expect the printing to be consistent – and for the rules it is, right? The example in the issue description demonstrates how the L010 violation is printed by both fix & lint.

That's because L010 is fixable by `fix`

[My example](https://github.com/sqlfluff/sqlfluff/issues/3372#issuecomment-1131482297) is for L013 - which is NOT fixable by `fix`. It only shows for `lint` but not for `fix`. When you run `fix` it just shows `Unfixable violations detected. ` but doesn't list that L014 is the unfixable error or where in your SQL it is.

PRS is the same - it's not a fixable error, hence does not shows in full in output and just says a summary (`1 templating/parsing errors found`)

So PRS is consistent with unfixable rules.

Do you get me?

**Comment by juhoautio** [comment]
Ok, I get your point. However when a PRS is also encountered, it makes the fixable violation effectively unfixable because nothing is done because of the PRS error (which is good :)).

**Comment by alanmcruickshank** [comment]
I'm going to close this issue because I think the desired functionality has now been implemented as https://docs.sqlfluff.com/en/stable/cli.html#cmdoption-sqlfluff-fix-show-lint-violations

**Comment by Myles1** [comment]
For anyone else who stumbles upon this through google, you're looking for the `--show-lint-violations` flag
(the above link is a 404 now)
